File: app/src/main/java/com/legue/axel/lolsummonertool/database/model/champion/Passive.java
package com.legue.axel.lolsummonertool.database.model.champion;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import static androidx.room.ForeignKey.CASCADE;

@Entity(tableName = "passives",
        foreignKeys = @ForeignKey(
                entity = Champion.class,
                parentColumns = "key",
                childColumns = "championId",
                onDelete = CASCADE
        ))
public class Passive {
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String name;
    public String description;
    public String image;
    @ColumnInfo(index = true)
    public int championId;

    @Ignore
    public Passive() {
    }

    public Passive(int id, String name, String description, int championId) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.championId = championId;
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/model/champion/Passive.kt
package com.legue.axel.lolsummonertool.database.model.champion

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.CASCADE
import androidx.room.PrimaryKey

@Entity(tableName = "passives", foreignKeys = [ForeignKey(entity = Champion::class, parentColumns = ["key"], childColumns = ["championId"], onDelete = CASCADE)])
data class Passive(
        @PrimaryKey(autoGenerate = true)
        val id: Int? = null,
        val name: String? = null,
        val description: String? = null,
        val image: String? = null,
        @ColumnInfo(index = true)
        val championId: Int = 0
)